nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/utils/NGSIUtils.java

```diff
@@ -166,7 +166,14 @@ private AttributesLD parseNgsiLdAttribute(String key, JSONObject value) {
                 if (value.get(keyOne) instanceof String)
                     subAttributes.add(new AttributesLD(keyOne.toLowerCase(), "Property", "", "", "", "", value.getString(keyOne), false, null));
                 else subAttributes.add(new AttributesLD(keyOne.toLowerCase(), "Property", "", "", "", "", null, false, null));
-
+            } else if ("GeoProperty".equals(attrType) && "value".equals(keyOne)) {
+                if(attrValue.equals("Point")) {
+                    String [] location = value.getJSONObject("value").get("coordinates").toString().replaceAll("\\[", "").replaceAll("\\]", "").split(",");
+                    subAttributes.add(new AttributesLD("lon", "GeoProperty", "", "", "", "", Double.parseDouble(location[0]), false, null));
+                    subAttributes.add(new AttributesLD("lat", "GeoProperty", "", "", "", "", Double.parseDouble(location[1]), false, null));
+                } else {
```

Comment:
It should not be an `else`. We may need extracted lat / lon **and** a GeoJSON / geometry representation.

Reply:
I didn't really understand what you said here

Reply:
if it is a point, we do something, in every case (including point) we do another thing.

Reply:
Ok I have understand